### PR TITLE
Dev memory allocate

### PR DESCRIPTION
**Pull request by fabiencastan**

### CODE REVIEW

File: libraries/tuttle/src/tuttle/host/memory/MemoryPool.cpp [hunk 1]
```diff
@@ -166,17 +178,44 @@ boost::intrusive_ptr<IPoolData> MemoryPool::allocate( const std::size_t size )
 
 	if( pData != NULL )
 	{
-		pData->_size = size;
```
Comment (fabiencastan) [inline libraries/tuttle/src/tuttle/host/memory/MemoryPool.cpp]: We need to update the "size" used (useful to compute the wasted size for example). The object still contains the real allocated size, called "reservedSize".

File: libraries/tuttle/src/tuttle/host/memory/MemoryCache.cpp [hunk 2]
```diff
@@ -47,6 +47,17 @@ CACHE_ELEMENT MemoryCache::get( const std::size_t& i ) const
 	return itr->second;
 }
 
+CACHE_ELEMENT MemoryCache::getUnusedWithSize( const std::size_t size ) const
+{
+	boost::mutex::scoped_lock lockerMap( _mutexMap );
+	for( MAP::const_iterator itr = _map.begin(); itr != _map.end(); ++itr )
+	{
+		if( size <= itr->second->getMemorySize() )
+			return itr->second;
```
Comment (fabiencastan) [inline libraries/tuttle/src/tuttle/host/memory/MemoryCache.cpp]: Need to find the unused element with the **smallest** size >= requestedSize.

Comment (fabiencastan) [inline libraries/tuttle/src/tuttle/host/memory/MemoryCache.cpp]: Need to find the **unused** elements:
`if( it->second->getReferenceCount( ofx::imageEffect::OfxhImage::eReferenceOwnerHost ) <= 1 )`
See `MemoryCache::clearUnused()`

Maybe better to create a function:
```
/**
 * @brief Check if the cache element is kept in the cache, but is not required by someone else.
 */
bool isUnused( const CACHE_ELEMENT& cacheElement )
{
    return cacheElement->getReferenceCount( ofx::imageEffect::OfxhImage::eReferenceOwnerHost ) <= 1;
}
```

**Comment by fabiencastan** [comment]
Could be great to re-enable the warning in the destructor (DEBUG level):
```
MemoryPool::~MemoryPool()
{
	if( !_dataUsed.empty() )
	{
		TUTTLE_LOG_DEBUG( "[Memory Pool] Error inside memory pool. Some data always mark used at the destruction (nb elements:" << _dataUsed.size() << ")" );
	}
}
```

**Comment by fabiencastan** [comment]
Could be great to add a constructor without maxSize.
`MemoryPool::MemoryPool()`
`MemoryPool::MemoryPool( const std::size_t maxSize )`

And add an enum to track if the user has set a maxSize manually. Based on this enum we could call updateMemoryAuthorizedWithRAM() in getMaxMemorySize() if we are in an automatic mode (if the delay with the last update is more than 1 min).

**Comment by cchampet** [comment]
Ping @fabiencastan

**Pull request merged**